src/python/pants/backend/jvm/tasks/jvmdoc_gen.py

```diff
@@ -111,7 +111,7 @@ def docable(target):
     if not targets:
       return
 
-    with self.invalidated(targets) as invalidation_check:
+    with self.invalidated(targets, invalidate_dependents=self.combined) as invalidation_check:
       def find_jvmdoc_targets():
         invalid_targets = set()
         for vt in invalidation_check.invalid_vts:
```

Comment:
I think that `if self.combined`, `find_jvmdoc_targets` should always return _all_ targets from the context... otherwise you might run on just a portion of the changed targets.